### PR TITLE
Replace cursor in OracleToPostgresCloner with custom pagination

### PR DESCRIPTION
**Pull request by JKoetsier**

### CODE REVIEW

File: src/gobprepare/cloner/oracle_to_postgres.py [hunk 1]
```diff
@@ -158,6 +159,60 @@ def _copy_data(self) -> int:
 
         return rows_copied
 
+    def _list_to_chunks(self, lst: list, chunk_size: int):
+        """Divides a list in chunks of chunk_size. Returns a list of dictionaries with "min" and "max" as keys, where
+        "min" is inclusive, and "max" is exclusive.
```
Comment (Vegter) [inline src/gobprepare/cloner/oracle_to_postgres.py]: Nice, as our Dutch Edsger Dijkstra already wrote in 1982,
https://www.cs.utexas.edu/users/EWD/ewd08xx/EWD831.PDF

File: src/gobprepare/cloner/oracle_to_postgres.py [hunk 2]
```diff
@@ -158,6 +159,60 @@ def _copy_data(self) -> int:
...
+
+        :param lst:
+        :param chunk_size:
+        :return:
+        """
+        result = []
+        min_idx = 0
+
+        while min_idx < len(lst):
+            row = {}
```
Comment (Vegter) [inline src/gobprepare/cloner/oracle_to_postgres.py]:
```suggestion
            max_idx = min_idx + chunk_size
            result.append({
                'min': lst[min_idx] if min_idx > 0 else None,
                'max': lst[max_idx] if max_idx < len(lst) else None
            }
            min_idx = max_idx
```
and then after the loop:

File: src/tests/gobprepare/cloner/test_oracle_to_postgres.py [hunk 3]
```diff
@@ -193,47 +193,126 @@ def test_copy_data(self):
...
         ])
 
+    def test_list_to_chunks(self):
+        testcases = [
+            (([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], 3), [{"min": None, "max": 4}, {"min": 4, "max": 7}, {"min": 7, "max": 10}, {"min": 10, "max": None}]),
+            (([1, 2, 3], 1), [{"min": None, "max": 2}, {"min": 2, "max": 3}, {"min": 3, "max": None}]),
+            (([], 4), []),
+            (([1, 3], 2), [{"min": None, "max": None}]),
+            (([1, 3], 5), [{"min": None, "max": None}]),
+        ]
```
Comment (Vegter) [inline src/tests/gobprepare/cloner/test_oracle_to_postgres.py]: probably hypothetical but what if the chunk_size = 0?

File: src/gobprepare/cloner/oracle_to_postgres.py [hunk 4]
```diff
@@ -185,33 +240,54 @@ def _copy_table_data(self, table_definition: Tuple[str, List]) -> int:
...
-        while True:
+
+        chunks = self._list_to_chunks(ids, self.READ_BATCH_SIZE)
+        order_field = self._get_id_columns_for_table(full_table_name)[0]
+
+        for chunk in chunks:
+            if chunk['min'] or chunk['max']:
+                where_max = f"{order_field} < '{chunk['max']}'" if chunk['max'] is not None else ""
+                where_min = f"{order_field} >= '{chunk['min']}'" if chunk['min'] is not None else ""
+                where = f"WHERE {where_min} {'AND' if where_min and where_max else ''} {where_max}"
```
Comment (Vegter) [inline src/gobprepare/cloner/oracle_to_postgres.py]: I find the number of None clauses a little confusing, but...
I tried for a long time to convert this in an 'AND '.join([...]) but didn't manage to produce any 'better' code.

File: src/tests/gobprepare/cloner/test_oracle_to_postgres.py (hunk 3 above)
Comment (JKoetsier) [reply src/tests/gobprepare/cloner/test_oracle_to_postgres.py]: Raises an error now :)

**Pull request merged**